test/test_eos_unit.cpp

```diff
@@ -193,14 +227,90 @@ SCENARIO("EOS Builder and Modifiers", "[EOSBuilder],[Modifiers][IdealGas]") {
         compare_two_eoss(igsh, ig);
         compare_two_eoss(igsc, ig);
+        compare_two_eoss(igra, ig);
+      }
+    }
+    WHEN("We construct a ramp from a p-alpha model") {
+      const Real Pe = 5.e7, Pc = 1.e8;
+      const Real alpha0 = 1.5;
+      const Real T0 = 293.0;
+      int enabled[4] = {0, 0, 0, 1};
+      Real vals[6] = {0.0, 0.0, alpha0, Pe, Pc, 0.0};
+      const Real rho0 = 1.e6 / (gm1 * Cv * T0);
+      EOS igra;
+      const Real r0 = rho0 / alpha0;
+      const Real r1 = Pc / (gm1 * Cv * T0);
+      const Real rmid = Pe / (gm1 * Cv * T0 * alpha0);
+      // P(alpha0 * rmid)
+      const Real P_armid = alpha0 * gm1 * Cv * rmid * T0;
+      init_sg_IdealGas(0, &igra, gm1, Cv, enabled, vals);
+      // construct ramp params and evaluate directly for test
+      const Real a = r0 * Pe / (rmid - r0);
+      const Real b = r0 * (Pc - Pe) / (r1 - rmid);
+      const Real c = (Pc * rmid - Pe * r1) / (r0 * (Pc - Pe));
+      // density in the middle of the first slope
+      const Real rho_t1 = 0.5 * (r0 + rmid);
+      // density in the middle of the second slope
+      const Real rho_t2 = 0.5 * (rmid + r1);
+      // P (rho_t1) note that r0 = rho0 / alpha0
+      const Real Prhot1 = a * (rho_t1 / r0 - 1.0);
+      // P (rho_t2)
+      const Real Prhot2 = b * (rho_t2 / r0 - c);
+      // bmod (rho_t1)
+      const Real bmodrt1 = rho_t1 * a / r0;
+      // bmod (rho_t2)
+      const Real bmodrt2 = rho_t2 * b / r0;
+      THEN("We obtain Pe and Pc when evaluating at mid and upper density values") {
+        REQUIRE(
+            isClose(P_armid, igra.PressureFromDensityTemperature(rmid, 293.0), 1.e-12));
+        // REQUIRE(isClose(Pc, igra.PressureFromDensityTemperature(r1, 293.0), 1.e-12));
+        // also check pressures on ramp
+        REQUIRE(isClose(Prhot1, igra.PressureFromDensityTemperature(rho_t1, T0), 1.e-12));
+        REQUIRE(isClose(Prhot2, igra.PressureFromDensityTemperature(rho_t2, T0), 1.e-12));
+        // check pressure below and beyond ramp matches unmodified ideal gas
+        REQUIRE(isClose(0.8 * r0 * gm1 * Cv * T0,
+                        igra.PressureFromDensityTemperature(0.8 * r0, T0), 1.e-12));
+        REQUIRE(isClose(1.2 * r1 * gm1 * Cv * T0,
+                        igra.PressureFromDensityTemperature(1.2 * r1, T0), 1.e-12));
+        // check bulk moduli on both pieces of ramp
+        REQUIRE(
+            isClose(bmodrt1, igra.BulkModulusFromDensityTemperature(rho_t1, T0), 1.e-12));
+        REQUIRE(
+            isClose(bmodrt2, igra.BulkModulusFromDensityTemperature(rho_t2, T0), 1.e-12));
+        // check bulk modulus below and beyond ramp matches unmodified ideal gas
+        REQUIRE(isClose(0.8 * r0 * gm1 * (gm1 + 1.0) * Cv * T0,
+                        igra.BulkModulusFromDensityTemperature(0.8 * r0, T0), 1.e-12));
+        REQUIRE(isClose(1.2 * r1 * gm1 * (gm1 + 1.0) * Cv * T0,
+                        igra.BulkModulusFromDensityTemperature(1.2 * r1, T0), 1.e-12));
```

Comment:
Can you break this up into multiple `THEN` statements? I think you just need to transform your comments into `THEN` statements so that you have five tests instead of one long one.

Reply:
done.